torchvision/csrc/cpu/image/readjpeg_cpu.cpp

```diff
@@ -100,15 +101,35 @@ torch::Tensor decodeJPEG(const torch::Tensor& data) {
 
   // read info from header.
   jpeg_read_header(&cinfo, TRUE);
+
+  int current_channels = cinfo.num_components;
```

Comment:
I see that we are now using `cinfo.num_components` instead of `cinfo.output_components`. Is there any particular difference / reason for this change?

Reply:
Good catch. My understanding is that `num_components` == `out_components` when you use it immediately after a `jpeg_read_header()`. The value of `out_components` changes after you call `jpeg_calc_output_dimensions()`. We also tested these changes with a variety of different assets and it works fine. Nevertheless, I agree that `output_components` requires less assumptions (maybe on the future libjpeg will do a mandatory conversion for some reason), so I'll replace it.

Reply:
Turns out the proposed change causes issues with damages pictures: https://app.circleci.com/pipelines/github/pytorch/vision/5000/workflows/21a22cd6-0f9f-459b-9c20-b2d1b0ff1c92/jobs/307793

Rolling `output_components ` back to `num_components`.